[wip] Backport Core #10344:Fix abandonconflict.py intermittency (jnewbery)

ftrader opened the pull request:
Had to add the new disconnect_nodes() method in util.py .
According to https://github.com/bitcoin/bitcoin/pull/10344/ :

"abandonconflict.py intermittently fails (about 1 time out of 25 for me). The failure is caused by a race between shutting down node0 and a transaction being INV'ed from node0 and getting into node1's mempool.

The original intent was that node0 and node1 should be disconnected before sending the transaction in node0. However, the call to disconnectnode() doesn't actually disconnect the nodes because there are two P2P connections between node0 and node1 (connect_nodes_bi() madness).

This PR change the direct call to disconnectnode() into a call to the test_framework/util function disconnect_nodes() which fully disconnects the nodes. It also asserts that node1's mempool is empty after stop-starting node0."

ftrader committed:
Backport Core #10344 Fix abandonconflict.py intermittency (jnewbery)

Had to add the new disconnect_nodes() method in util.py .
According to https://github.com/bitcoin/bitcoin/pull/10344/ :

"abandonconflict.py intermittently fails (about 1 time out of 25 for me). The failure is caused by a race between shutting down node0 and a transaction being INV'ed from node0 and getting into node1's mempool.

The original intent was that node0 and node1 should be disconnected before sending the transaction in node0. However, the call to disconnectnode() doesn't actually disconnect the nodes because there are two P2P connections between node0 and node1 (connect_nodes_bi() madness).

This PR change the direct call to disconnectnode() into a call to the test_framework/util function disconnect_nodes() which fully disconnects the nodes. It also asserts that node1's mempool is empty after stop-starting node0."

sickpig commented:
@ftrader travis failed on `abandonconflict.py` (linux 32bit) 
https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/232530991#L3257

ftrader commented:
To preserve the Travis error:
```
[301, 301]
start_node: bitcoind started, waiting for RPC to come up
start_node: RPC succesfully started
Assertion failed: 2 != 0
  File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/qa/rpc-tests/test_framework/test_framework.py", line 184, in main
    self.run_test()
  File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/build/../qa/rpc-tests/abandonconflict.py", line 78, in run_test
    assert_equal(len(self.nodes[1].getrawmempool()), 0)
  File "/home/travis/build/BitcoinUnlimited/BitcoinUnlimited/qa/rpc-tests/test_framework/util.py", line 612, in assert_equal
    raise AssertionError("%s != %s"%(str(thing1),str(thing2)))
```

sickpig commented:
@ftrader testing locally. With this patch applied I have a significantly higher failure rate 2 out of 15, whereas in the latest experiment with `dev` @  76fd9a6 I got 1 failure in 80 runs.

ftrader committed:
Backport util.py changes to default start_node arguments

I tested these in following way:

- normal regtest suite passed completely
- extended regtests that failed were checked against 'dev' without PR582

The extended tests that failed seemed to fail without this PR already. 
They were:

- bip9-softforks (crash)
- bip68-sequence (loops)
- bipdersig-p2p (crash)
- mempool_packages (loops)
- p2p-acceptblock (intermittent failure)

ftrader commented:
@sickpig : based on your Slack feedback the patch may not have been effective without the 'uacomment' argument added in last commit to util.py, which I overlooked in my initial backporting.

May I ask you to do some more runs with that commit included? Thanks.

ftrader commented:
Latest Travis failure was, unfortunately, a neverending loop on abandonconflict.py on a single test platform.

https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/232806193
